src/devices/roccat_elo_7_1_air.c

```diff
@@ -0,0 +1,97 @@
+#include "../device.h"
+
+#include <stdio.h>
+#include <string.h>
+#include <time.h>
+
+static struct device device_elo71Air;
+
+static const uint16_t PRODUCT_ID = 0x3a37;
+
+static int elo71Air_switch_lights(hid_device* hid_device, uint8_t on);
+
+void elo71Air_init(struct device** device)
+{
+    device_elo71Air.idVendor = VENDOR_ROCCAT;
+    device_elo71Air.idProductsSupported = &PRODUCT_ID;
+    device_elo71Air.numIdProducts = 2;
+    device_elo71Air.idInterface = 0;
+
+    strncpy(device_elo71Air.device_name, "ROCCAT Elo 7.1 Air", sizeof(device_elo71Air.device_name));
+
+    device_elo71Air.capabilities = CAP_LIGHTS;
+    device_elo71Air.switch_lights = &elo71Air_switch_lights;
+
+    *device = &device_elo71Air;
+}
+
+/*
+ * Headset exchanges 64 Byte blocks with mostly zeros with driver. => This wrapper
+ * around hid_write and hid_read pads zeros between the passed, variable length
+ * data (length specified by size) and 64. If out_buffer points to 64 byte of 
+ * preallocated memory, send_receive will receive 64 byte into id over the HID
+ * stream.
+ *
+ * The Roccat Elo 71 Air hates, if it gets 64 bytes blocks to quickly. The 
+ * windows software seems just to wait a bit, so send_receive does as well. :-/
+ *
+ * This function returns the code from hid_read or hid_write, if either no read
+ * was requested or hid_write already failed.
+ */
+static int send_receive(hid_device* hid_device, const unsigned char* data, 
+			int size, unsigned char* out_buffer) 
+{
+    struct timespec ts;	
+    unsigned char send_buffer[64];
+    int r;
+
+    memset(send_buffer, 0, sizeof(send_buffer));
+    memcpy(send_buffer, data, size);
```

Comment:
Thank you!

Just a question regarding your send_receive function. As I dont quite understand the sense.

E.g. when you do this:
```
     unsigned char cmd92[64] = {0xff, 0x02};

     hid_write(hid_device, cmd92, sizeof(cmd92)/sizeof(cmd92[0]));
```

The first line, will automatically pad everything with 0s. Thats defined in the C-Standard. So you dont actually have to clear the array with memset beforehand.

Reply:
Thank you for the feedback!

If that's defined in the C-Standard, I'll of course remove this as obsolete.

It is quite a time ago that I actively programmed C / Assembler and just remembered, that I should rather be defensive about such things. ;-)

Reply:
I hope that I corrected the code properly.